tests/testthat/test-assert.R

```diff
@@ -238,6 +269,59 @@ test_that("names and row names do not influence type identity (#707)", {
   expect_true(vec_is(mtcars, structure(mtcars, row.names = 1:32)))
 })
 
+# vec_check_size --------------------------------------------------------
+
+test_that("vec_check_size() is silent if the size is right", {
+  expect_null(vec_check_size(1:5, size = 5L))
+  expect_null(vec_check_size(data_frame(.size = 10L), size = 10L))
+})
+
+test_that("vec_check_size() errors on the wrong size", {
+  expect_snapshot(error = TRUE, {
+    vec_check_size(1:5, size = 1L)
+  })
+  expect_snapshot(error = TRUE, {
+    vec_check_size(1:5, size = 10L)
+  })
+})
+
+test_that("vec_check_size() errors on scalars", {
```

Comment:
This returns a good enough error on scalars that I think

```r
vec_assert(x, size = 5)
```

can be replaced with just

```r
vec_check_size(x, size = 5)
```

rather than having to do both of these

```r
vec_check_vector(x)
vec_check_size(x, size = 5)
```

which is great